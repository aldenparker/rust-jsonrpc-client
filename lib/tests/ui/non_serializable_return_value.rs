struct Number;

#[jsonrpc_client::api]
pub trait Math {
    async fn subtract(&self, subtrahend: i64, minuend: i64) -> Number;
}

fn main() {}
