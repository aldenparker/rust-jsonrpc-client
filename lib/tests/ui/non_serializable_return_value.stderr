error[E0277]: the trait bound `for<'de> Number: serde::Deserialize<'de>` is not satisfied
 --> $DIR/non_serializable_return_value.rs:5:64
  |
5 |     async fn subtract(&self, subtrahend: i64, minuend: i64) -> Number;
  |                                                                ^^^^^^ the trait `for<'de> serde::Deserialize<'de>` is not implemented for `Number`
  |
  = note: required because of the requirements on the impl of `serde::de::DeserializeOwned` for `Number`
